### ISSUE TITLE
m.request post headers options，Is the Content-Type here is normal？

### ISSUE THREAD

**Issue by stylering** [MithrilJS/mithril.js#1732]
```js
m.request('http://host/url', {
     methods: 'POST',
     data: {a: 1, b: 1},
     headers: {
           'Content-Type': 'application/x-www-form-urlencoded'
     ]
}
```

in chrome Request Headers result：

```
Content-Type:application/json; charset=UTF-8 application/x-www-form-urlencoded
```

Is the Content-Type here is normal？

**Comment by volnei** [comment]
You must use deserialize function, by default mithril will check if deserealize function is equal to JSON.stringfy and use application/json header.

```javascript  
m.request('http://host/url', {
    methods: 'POST',
    data: {a: 1, b: 1},
    headers: {
    "Content-Type': 'application/x-www-form-urlencoded'
    ],
    deserealize: function(data) {return data}
}
```

Regards

**Comment by isiahmeadows** [comment]
@stylering That's odd in that it's being appended to (it should be replaced or ignored), but if you're wanting to send something with that content type, try `serialize: m.route.buildQueryString`. It might look like this:

```js
m.request('http://host/url', {
     methods: 'POST',
     data: {a: 1, b: 1},
     serialize: m.route.buildQueryString,
     headers: {
           'Content-Type': 'application/x-www-form-urlencoded'
     }
})
```

Also, I fixed your initial comment's formatting, but I left the erroneous `]` in your `headers` parameter and the missing `)` in your `m.request` call.

**Comment by isiahmeadows** [comment]
By default, `m.request` issues a JSON request, absent a `serialize` and/or `deserialize` that isn't `JSON.stringify`, and it sets the headers accordingly when it does.

**Comment by isiahmeadows** [comment]
Fixed in v1.1.6, so closing.
